tests/unit_tests/test_matchmaker_algorithm_bucket_teams.py

```diff
@@ -214,26 +219,33 @@ def test_distribute_pairs_1(player_factory):
 
 
 def test_distribute_pairs_2(player_factory):
-    players = [player_factory(1500, 500, name=f"p{i+1}") for i in range(8)]
+    players = [
+        player_factory(1500, 500, name=f"p{i+1}", player_id=i+1)
+        for i in range(8)
+    ]
     searches = [(Search([player]), 0) for player in players]
     p1, p2, p3, p4, p5, p6, p7, p8 = players
 
     grouped = [
         search.players for search in algorithm.bucket_teams._distribute(searches, 2)
     ]
-    assert grouped == [[p1, p4], [p2, p3], [p5, p8], [p6, p7]]
+    assert grouped == [[p1, p7], [p2, p8], [p3, p5], [p4, p6]]
```

Comment:
Hilariously, these tests never actually worked properly since all player objects had the same id and therefore compared equal. They were only able to assert that the returned list had the correct shape.